test/nullable.jl

```diff
@@ -388,6 +392,77 @@ end
 @test Base.promote_op(-, Nullable{Float64}, Nullable{Int}) == Nullable{Float64}
 @test Base.promote_op(-, Nullable{DateTime}, Nullable{DateTime}) == Nullable{Base.Dates.Millisecond}
 
+# filter
+for p in (_ -> true, _ -> false)
+    @test @inferred(filter(p, Nullable()))      |> isnull_oftype(Union{})
+    @test @inferred(filter(p, Nullable{Int}())) |> isnull_oftype(Int)
+end
+@test @inferred(filter(_ -> true, Nullable(85)))  === Nullable(85)
+@test @inferred(filter(_ -> false, Nullable(85))) |> isnull_oftype(Int)
+@test @inferred(filter(x -> x > 0, Nullable(85))) === Nullable(85)
+@test @inferred(filter(x -> x < 0, Nullable(85))) |> isnull_oftype(Int)
+@test get(@inferred(filter(x -> length(x) > 2, Nullable("test")))) == "test"
+@test @inferred(filter(x -> length(x) > 5, Nullable("test"))) |>
+    isnull_oftype(String)
+
+# map
+sqr(x) = x^2
+@test @inferred(map(sqr, Nullable()))        |> isnull_oftype(Union{})
+@test @inferred(map(sqr, Nullable{Int}()))   |> isnull_oftype(Int)
+@test @inferred(map(sqr, Nullable(2)))       === Nullable(4)
+@test @inferred(map(+, Nullable(0.0)))       === Nullable(0.0)
+@test @inferred(map(-, Nullable(1.0)))       === Nullable(-1.0)
+@test @inferred(map(-, Nullable{Float64}())) |> isnull_oftype(Float64)
+@test @inferred(map(sin, Nullable(1)))       === Nullable(sin(1))
+@test @inferred(map(sin, Nullable{Int}()))   |> isnull_oftype(Float64)
+
+# should not throw if function wouldn't be called
+@test map(x -> x ? 0 : 0.0, Nullable())       |> isnull_oftype(Union{})
+@test map(x -> x ? 0 : 0.0, Nullable(true))   === Nullable(0)
+@test map(x -> x ? 0 : 0.0, Nullable(false))  === Nullable(0.0)
+@test map(x -> x ? 0 : 0.0, Nullable{Bool}()) |> isnull_oftype(Union{})
+
+# broadcast and elementwise
+# @inferred cannot be used on these because sin.() is not a call expression
```

Comment:
I'd rather `@inferred` be changed so that this syntax can be used, if possible. Will look into that and possibly open a separate PR. Otherwise, we can just use `broadcast` and assume that dot operators are tested elsewhere.